From: Jean Delvare <jdelvare@suse.de>
Date: Wed, 16 Jan 2019 09:04:55 +0100
Subject: dmidecode: Add "Logical non-volatile device" to the memory device types
Git-commit: 74dfb854b8199ddb0a27e89296fa565f4706cb9d
Patch-mainline: yes
References: bsc#1120149

When adding support for non-volative memory, we forgot to add
"Logical non-volatile device" to the list of memory types. This
causes NVDIMM modules to show up as <OUT OF SPEC>. Fix the problem
by adding the missing enumerated value.

Signed-off-by: Jean Delvare <jdelvare@suse.de>
Reviewed-by: Jerry Hoemann <jerry.hoemann@hpe.com>
---
 dmidecode.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/dmidecode.c
+++ b/dmidecode.c
@@ -2469,10 +2469,11 @@ static const char *dmi_memory_device_typ
 		"LPDDR",
 		"LPDDR2",
 		"LPDDR3",
-		"LPDDR4" /* 0x1E */
+		"LPDDR4",
+		"Logical non-volatile device" /* 0x1F */
 	};
 
-	if (code >= 0x01 && code <= 0x1E)
+	if (code >= 0x01 && code <= 0x1F)
 		return type[code - 0x01];
 	return out_of_spec;
 }
